### PR TITLE
sqs: Fix diffs with policies

### LINKED ISSUE #19836: tests/resource/aws_sqs_queue: GovCloud eventual consistency errors
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

Relates: #19834.

More details to follow...

### LINKED ISSUE #20333: aws_sqs_queue_policy error reading, empty result
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

```
Terraform v0.13.5
+ provider registry.terraform.io/hashicorp/aws v3.50.0
```

### Affected Resource(s)

* aws_sqs_queue_policy

### Terraform Configuration Files

```hcl
resource "aws_sqs_queue" "this" {
  name = "test"
}

resource "aws_sqs_queue_policy" "this" {
  queue_url = aws_sqs_queue.this.id
  policy    = data.aws_iam_policy_document.this.json
}

data "aws_iam_policy_document" "this" {
  policy_id = aws_sqs_queue.this.arn

  statement {
    actions = [
      "SQS:*",
    ]
    resources = [
      aws_sqs_queue.this.arn,
    ]
    effect = "Allow"

    principals {
      identifiers = [
        "*",
      ]
      type = "*"
    }
  }
}
```

### Debug Output

Snippet for a failed apply of the `aws_sqs_queue_policy` resource:

```
2021-07-26T18:11:31.485Z [INFO]  plugin.terraform-provider-aws_v3.50.0_x5: 2021/07/26 18:11:31 [DEBUG] [aws-sdk-go] DEBUG: Request sqs/GetQueueAttributes Details:
---[ REQUEST POST-SIGN ]-----------------------------
POST / HTTP/1.1
Host: sqs.us-west-2.amazonaws.com
...
Action=GetQueueAttributes&AttributeName.1=All&QueueUrl=https%3A%2F%2Fsqs.us-west-2.amazonaws.com%2Fxxx%2Ftest&Version=2012-11-05
...
2021-07-26T18:11:31.529Z [INFO]  plugin.terraform-provider-aws_v3.50.0_x5: 2021/07/26 18:11:31 [DEBUG] [aws-sdk-go] DEBUG: Response sqs/GetQueueAttributes Details:
---[ RESPONSE ]--------------------------------------
HTTP/1.1 200 OK
...
-----------------------------------------------------: timestamp=2021-07-26T18:11:31.529Z
2021-07-26T18:11:31.529Z [INFO]  plugin.terraform-provider-aws_v3.50.0_x5: 2021/07/26 18:11:31 [DEBUG] [aws-sdk-go] <?xml version="1.0"?><GetQueueAttributesResponse xmlns="http://queue.amazonaws.com/doc/2012-11-05/"><GetQueueAttributesResult/><ResponseMetadata><RequestId>xxx</RequestId></ResponseMetadata></GetQueueAttributesResponse>: timestamp=2021-07-26T18:11:31.529Z
2021/07/26 18:11:31 [DEBUG] aws_sqs_queue_policy.this: apply errored, but we're indicating that via the Error pointer rather than returning it: error reading SQS Queue Policy (https://sqs.us-west-2.amazonaws.com/xxx/test): Empty result
2021/07/26 18:11:31 [TRACE] eval: *terraform.EvalMaybeTainted
2021/07/26 18:11:31 [TRACE] EvalMaybeTainted: aws_sqs_queue_policy.test encountered an error during creation, so it is now marked as tainted
```

Snippet for a successful apply of the `aws_sqs_queue_policy` resource:

```
2021-07-27T14:24:50.958Z [INFO]  plugin.terraform-provider-aws_v3.50.0_x5: 2021/07/27 14:24:50 [DEBUG] [aws-sdk-go] DEBUG: Response sqs/GetQueueAttributes Details:
---[ RESPONSE ]--------------------------------------
HTTP/1.1 200 OK
...

-----------------------------------------------------: timestamp=2021-07-27T14:24:50.958Z
2021-07-27T14:24:50.958Z [INFO]  plugin.terraform-provider-aws_v3.50.0_x5: 2021/07/27 14:24:50 [DEBUG] [aws-sdk-go] <?xml version="1.0"?><GetQueueAttributesResponse xmlns="http://queue.amazonaws.com/doc/2012-11-05/"><GetQueueAttributesResult><Attribute><Name>Policy</Name><Value>{&quot;Version&quot;:&quot;2012-10-1
7&quot;,&quot;Id&quot;:&quot;arn:aws:sqs:us-west-2:xxx:test&quot;,&quot;Statement&quot;:[{&quot;Sid&quot;:&quot;&quot;,&quot;Effect&quot;:&quot;Allow&quot;,&quot;Principal&quot;:&quot;*&quot;,&quot;Action&quot;:&quot;SQS:*&quot;,&quot;Resource&quot;:&quot;arn:aws:sqs:
us-west-2:xxx:test&quot;,&quot;Condition&quot;:{&quot;ArnLike&quot;:{&quot;aws:SourceArn&quot;:&quot;arn:aws:sns:us-west-2:xxx:test&quot;}}}]}</Value></Attribute></GetQueueAttributesResult><ResponseMetadata><RequestId>xxx</RequestId></ResponseMetadata></GetQueueAttributesResponse>: timestamp=2021-07-27T14:24:50.958Z
```

### Expected Behavior

`aws_sqs_queue_policy` resource should have been created properly and the apply should have succeeded.

### Actual Behavior

Most often, the expected behavior occurs. Intermittently, though, Terraform displayed the following error:

```
Error: error reading SQS Queue Policy (https://sqs.us-west-2.amazonaws.com/xxx/test): Empty result
```

### Steps to Reproduce

`terraform apply`

### Important Factoids

After the error has occurred, the apply can be repeated and successfully create the resource. We had not seen the issue prior to upgrading to AWS provider version 3.50.0. We had previously been using AWS provider version 3.4.0, for several months. On 3.50.0, we have now seen this issue 5 times or so out of a few dozen applies.

### References

* #19836 - This issue is likely the same. I only reported this issue separately to indicate that we're seeing it in production deploys, not just with the provider tests.

### PR DESCRIPTION
**Pull request by YakDriver**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates #21968
Fixes #19836
Fixes #20333

**NOTE:** These changes will make SQS configurations take a little longer to apply. However, because SQS has such trouble with eventual consistency, the extra time is needed to improve consistency.

I ran each test 10X in 3 different regions. Unfortunately, these changes _do not_ perfectly fix all eventual consistency problems (see #19836 and #20333). However, the overall improvement is significant:

| Test | GovCloud | `us-west-2` | `us-east-1` |
| --- | --- | --- | --- |
| **Total** Out of 260 (before --> after) | 182 --> 258 | 255 --> 259 | 258 --> 258 |
| TestAccSQSQueue_basic | 9 --> 10| 10 --> 10 | 10 --> 10 |
| TestAccSQSQueue_defaultKMSDataKeyReusePeriodSeconds | 6 --> 10| 10 --> 10 | 10 --> 10 |
| TestAccSQSQueue_disappears | 8 --> 10| 10 --> 9 | 10 --> 10 |
| TestAccSQSQueue_encryption | 2 --> 10| 9 --> 10 | 10 --> 10 |
| TestAccSQSQueue_fifoQueue | 7 --> 10| 10 --> 10 | 10 --> 10 |
| TestAccSQSQueue_FIFOQueue_contentBasedDeduplication | 8 --> 10| 10 --> 10 | 10 --> 10 |
| TestAccSQSQueue_FIFOQueue_expectNameError | 10 --> 10| 10 --> 10 | 10 --> 10 |
| TestAccSQSQueue_FIFOQueue_highThroughputMode | 6 --> 10| 9 --> 10 | 10 --> 10 |
| TestAccSQSQueue_Name_generated | 9 --> 10| 10 --> 10 | 10 --> 9 |
| TestAccSQSQueue_NameGenerated_fifoQueue | 7 --> 10| 10 --> 10 | 10 --> 10 |
| TestAccSQSQueue_namePrefix | 9 --> 10| 10 --> 10 | 10 --> 10 |
| TestAccSQSQueue_NamePrefix_fifoQueue | 10 --> 10| 10 --> 10 | 10 --> 10 |
| TestAccSQSQueue_Policy_basic | 10 --> 10| 10 --> 10 | 10 --> 10 |
| TestAccSQSQueue_Policy_ignoreEquivalent | 10 --> 10| 10 --> 10 | 10 --> 10 |
| TestAccSQSQueue_recentlyDeleted | 7 --> 10| 10 --> 10 | 10 --> 10 |
| TestAccSQSQueue_redrivePolicy | 7 --> 10| 9 --> 10 | 10 --> 10 |
| TestAccSQSQueue_StandardQueue_expectContentBased... | 10 --> 10| 10 --> 10 | 10 --> 10 |
| TestAccSQSQueue_tags | 6 --> 10| 9 --> 10 | 10 --> 10 |
| TestAccSQSQueue_update | 3 --> 10| 10 --> 10 | 10 --> 10 |
| TestAccSQSQueue_zeroVisibilityTimeoutSeconds | 7 --> 10| 10 --> 10 | 10 --> 9 |
| TestAccSQSQueueDataSource_basic | 6 --> 9 | 10 --> 10 | 10 --> 10 |
| TestAccSQSQueueDataSource_tags | 7 --> 10| 10 --> 10 | 10 --> 10 |
| TestAccSQSQueuePolicy_basic | 2 --> 10| 9 --> 10 | 9 --> 10 |
| TestAccSQSQueuePolicy_disappears | 3 --> 10| 10 --> 10 | 10 --> 10 |
| TestAccSQSQueuePolicy_Disappears_queue | 8 --> 9| 10 --> 10 | 10 --> 10 |
| TestAccSQSQueuePolicy_update | 5 --> 10| 10 --> 10 | 9 --> 10 |

Output from acceptance testing (`us-west-2`):

```console
% make testacc TESTS=TestAccSQSQueue PKG=sqs
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/sqs/... -v -count 1 -parallel 20 -run='TestAccSQSQueue' -timeout 180m
--- PASS: TestAccSQSQueue_FIFOQueue_expectNameError (5.24s)
--- PASS: TestAccSQSQueue_StandardQueue_expectContentBasedDeduplicationError (5.43s)
--- PASS: TestAccSQSQueue_NamePrefix_fifoQueue (49.23s)
--- PASS: TestAccSQSQueueDataSource_basic (49.52s)
--- PASS: TestAccSQSQueue_fifoQueue (52.91s)
--- PASS: TestAccSQSQueue_zeroVisibilityTimeoutSeconds (53.05s)
--- PASS: TestAccSQSQueue_FIFOQueue_contentBasedDeduplication (53.19s)
--- PASS: TestAccSQSQueue_NameGenerated_fifoQueue (53.24s)
--- PASS: TestAccSQSQueue_basic (53.24s)
--- PASS: TestAccSQSQueue_defaultKMSDataKeyReusePeriodSeconds (53.31s)
--- PASS: TestAccSQSQueue_namePrefix (53.33s)
--- PASS: TestAccSQSQueue_Name_generated (53.33s)
--- PASS: TestAccSQSQueue_disappears (58.84s)
--- PASS: TestAccSQSQueue_redrivePolicy (67.99s)
--- PASS: TestAccSQSQueue_tags (75.07s)
--- PASS: TestAccSQSQueue_update (79.35s)
--- PASS: TestAccSQSQueue_FIFOQueue_highThroughputMode (79.66s)
--- PASS: TestAccSQSQueueDataSource_tags (37.40s)
--- PASS: TestAccSQSQueuePolicy_Disappears_queue (90.95s)
--- PASS: TestAccSQSQueue_Policy_ignoreEquivalent (93.70s)
--- PASS: TestAccSQSQueuePolicy_update (102.57s)
--- PASS: TestAccSQSQueue_encryption (104.95s)
--- PASS: TestAccSQSQueuePolicy_basic (73.36s)
--- PASS: TestAccSQSQueue_recentlyDeleted (123.80s)
--- PASS: TestAccSQSQueuePolicy_disappears (83.69s)
--- PASS: TestAccSQSQueue_Policy_basic (87.59s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/sqs	138.023s
```

Output from acceptance testing (GovCloud):

```console
% make testacc TESTS=TestAccSQSQueue PKG=sqs
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/sqs/... -v -count 1 -parallel 20 -run='TestAccSQSQueue' -timeout 180m
--- PASS: TestAccSQSQueue_StandardQueue_expectContentBasedDeduplicationError (6.27s)
--- PASS: TestAccSQSQueue_FIFOQueue_expectNameError (6.43s)
--- PASS: TestAccSQSQueue_fifoQueue (51.83s)
--- PASS: TestAccSQSQueue_Name_generated (75.05s)
--- PASS: TestAccSQSQueue_Policy_ignoreEquivalent (100.83s)
--- PASS: TestAccSQSQueue_Policy_basic (95.58s)
--- PASS: TestAccSQSQueue_FIFOQueue_contentBasedDeduplication (103.79s)
--- PASS: TestAccSQSQueue_zeroVisibilityTimeoutSeconds (105.19s)
--- PASS: TestAccSQSQueue_basic (112.31s)
--- PASS: TestAccSQSQueue_disappears (116.04s)
--- PASS: TestAccSQSQueue_defaultKMSDataKeyReusePeriodSeconds (117.01s)
--- PASS: TestAccSQSQueue_redrivePolicy (127.00s)
--- PASS: TestAccSQSQueuePolicy_update (134.62s)
--- PASS: TestAccSQSQueuePolicy_basic (139.97s)
--- PASS: TestAccSQSQueuePolicy_disappears (143.00s)
--- PASS: TestAccSQSQueuePolicy_Disappears_queue (146.17s)
--- PASS: TestAccSQSQueueDataSource_tags (153.68s)
--- PASS: TestAccSQSQueueDataSource_basic (158.34s)
--- PASS: TestAccSQSQueue_update (154.88s)
--- PASS: TestAccSQSQueue_FIFOQueue_highThroughputMode (162.26s)
--- PASS: TestAccSQSQueue_encryption (164.73s)
--- PASS: TestAccSQSQueue_NamePrefix_fifoQueue (102.71s)
--- PASS: TestAccSQSQueue_namePrefix (97.73s)
--- PASS: TestAccSQSQueue_NameGenerated_fifoQueue (103.07s)
--- PASS: TestAccSQSQueue_recentlyDeleted (210.20s)
--- PASS: TestAccSQSQueue_tags (182.22s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/sqs	235.349s
```

### CODE REVIEW

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTS=TestAccSQSQueue_ PKG=sqs
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/sqs/... -v -count 1 -parallel 20 -run='TestAccSQSQueue_' -timeout 180m
=== RUN   TestAccSQSQueue_basic
=== PAUSE TestAccSQSQueue_basic
=== RUN   TestAccSQSQueue_disappears
=== PAUSE TestAccSQSQueue_disappears
=== RUN   TestAccSQSQueue_Name_generated
=== PAUSE TestAccSQSQueue_Name_generated
=== RUN   TestAccSQSQueue_NameGenerated_fifoQueue
=== PAUSE TestAccSQSQueue_NameGenerated_fifoQueue
=== RUN   TestAccSQSQueue_namePrefix
=== PAUSE TestAccSQSQueue_namePrefix
=== RUN   TestAccSQSQueue_NamePrefix_fifoQueue
=== PAUSE TestAccSQSQueue_NamePrefix_fifoQueue
=== RUN   TestAccSQSQueue_tags
=== PAUSE TestAccSQSQueue_tags
=== RUN   TestAccSQSQueue_update
=== PAUSE TestAccSQSQueue_update
=== RUN   TestAccSQSQueue_Policy_basic
=== PAUSE TestAccSQSQueue_Policy_basic
=== RUN   TestAccSQSQueue_Policy_ignoreEquivalent
=== PAUSE TestAccSQSQueue_Policy_ignoreEquivalent
=== RUN   TestAccSQSQueue_recentlyDeleted
=== PAUSE TestAccSQSQueue_recentlyDeleted
=== RUN   TestAccSQSQueue_redrivePolicy
=== PAUSE TestAccSQSQueue_redrivePolicy
=== RUN   TestAccSQSQueue_fifoQueue
=== PAUSE TestAccSQSQueue_fifoQueue
=== RUN   TestAccSQSQueue_FIFOQueue_expectNameError
=== PAUSE TestAccSQSQueue_FIFOQueue_expectNameError
=== RUN   TestAccSQSQueue_FIFOQueue_contentBasedDeduplication
=== PAUSE TestAccSQSQueue_FIFOQueue_contentBasedDeduplication
=== RUN   TestAccSQSQueue_FIFOQueue_highThroughputMode
=== PAUSE TestAccSQSQueue_FIFOQueue_highThroughputMode
=== RUN   TestAccSQSQueue_StandardQueue_expectContentBasedDeduplicationError
=== PAUSE TestAccSQSQueue_StandardQueue_expectContentBasedDeduplicationError
=== RUN   TestAccSQSQueue_encryption
=== PAUSE TestAccSQSQueue_encryption
=== RUN   TestAccSQSQueue_zeroVisibilityTimeoutSeconds
=== PAUSE TestAccSQSQueue_zeroVisibilityTimeoutSeconds
=== RUN   TestAccSQSQueue_defaultKMSDataKeyReusePeriodSeconds
=== PAUSE TestAccSQSQueue_defaultKMSDataKeyReusePeriodSeconds
=== CONT  TestAccSQSQueue_basic
=== CONT  TestAccSQSQueue_redrivePolicy
=== CONT  TestAccSQSQueue_StandardQueue_expectContentBasedDeduplicationError
=== CONT  TestAccSQSQueue_FIFOQueue_highThroughputMode
=== CONT  TestAccSQSQueue_encryption
=== CONT  TestAccSQSQueue_disappears
=== CONT  TestAccSQSQueue_FIFOQueue_contentBasedDeduplication
=== CONT  TestAccSQSQueue_FIFOQueue_expectNameError
=== CONT  TestAccSQSQueue_fifoQueue
=== CONT  TestAccSQSQueue_tags
=== CONT  TestAccSQSQueue_recentlyDeleted
=== CONT  TestAccSQSQueue_Policy_ignoreEquivalent
=== CONT  TestAccSQSQueue_Policy_basic
=== CONT  TestAccSQSQueue_update
=== CONT  TestAccSQSQueue_NameGenerated_fifoQueue
=== CONT  TestAccSQSQueue_NamePrefix_fifoQueue
=== CONT  TestAccSQSQueue_zeroVisibilityTimeoutSeconds
=== CONT  TestAccSQSQueue_namePrefix
=== CONT  TestAccSQSQueue_defaultKMSDataKeyReusePeriodSeconds
=== CONT  TestAccSQSQueue_Name_generated
--- PASS: TestAccSQSQueue_StandardQueue_expectContentBasedDeduplicationError (7.66s)
--- PASS: TestAccSQSQueue_FIFOQueue_expectNameError (8.02s)
--- PASS: TestAccSQSQueue_disappears (31.96s)
--- PASS: TestAccSQSQueue_zeroVisibilityTimeoutSeconds (40.56s)
--- PASS: TestAccSQSQueue_defaultKMSDataKeyReusePeriodSeconds (44.24s)
--- PASS: TestAccSQSQueue_Name_generated (44.38s)
--- PASS: TestAccSQSQueue_NameGenerated_fifoQueue (44.38s)
--- PASS: TestAccSQSQueue_basic (44.40s)
--- PASS: TestAccSQSQueue_NamePrefix_fifoQueue (44.45s)
--- PASS: TestAccSQSQueue_FIFOQueue_contentBasedDeduplication (44.63s)
--- PASS: TestAccSQSQueue_fifoQueue (44.68s)
--- PASS: TestAccSQSQueue_namePrefix (44.80s)
--- PASS: TestAccSQSQueue_redrivePolicy (47.10s)
--- PASS: TestAccSQSQueue_update (55.88s)
--- PASS: TestAccSQSQueue_FIFOQueue_highThroughputMode (56.22s)
--- PASS: TestAccSQSQueue_tags (63.90s)
--- PASS: TestAccSQSQueue_encryption (67.10s)
--- PASS: TestAccSQSQueue_Policy_ignoreEquivalent (78.01s)
--- PASS: TestAccSQSQueue_Policy_basic (84.02s)
--- PASS: TestAccSQSQueue_recentlyDeleted (92.32s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/sqs	95.919s
```

File: internal/service/sqs/queue.go [hunk 1]
```diff
@@ -272,7 +284,30 @@ func resourceQueueRead(d *schema.ResourceData, meta interface{}) error {
...
+		return nil
+	})
+
+	if tfresource.TimedOut(err) {
+		output, err = FindQueueAttributesByURL(conn, d.Id())
+	}
+
+	if err != nil {
+		return fmt.Errorf("error reading SQS Queue (%s): %w", d.Id(), err)
+	}
```
Comment (ewbankkit) [inline internal/service/sqs/queue.go:310]: Remove (there is an identical code block below), else:

```console
% make testacc TESTS=TestAccSQSQueue_disappears PKG=sqs
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/sqs/... -v -count 1 -parallel 20 -run='TestAccSQSQueue_disappears' -timeout 180m
=== RUN   TestAccSQSQueue_disappears
=== PAUSE TestAccSQSQueue_disappears
=== CONT  TestAccSQSQueue_disappears
    queue_test.go:70: Step 1/1 error: Error running post-apply refresh: exit status 1
        
        Error: error reading SQS Queue (https://sqs.us-west-2.amazonaws.com/187416307283/tf-acc-test-930144652419390609): couldn't find resource
        
          with aws_sqs_queue.test,
          on terraform_plugin_test.tf line 2, in resource "aws_sqs_queue" "test":
           2: resource "aws_sqs_queue" "test" {
        
--- FAIL: TestAccSQSQueue_disappears (46.53s)
FAIL
FAIL	github.com/hashicorp/terraform-provider-aws/internal/service/sqs	49.682s
FAIL
make: *** [testacc] Error 1
```

File: internal/service/sqs/queue.go [hunk 2]
```diff
@@ -272,7 +284,30 @@ func resourceQueueRead(d *schema.ResourceData, meta interface{}) error {
...
+		if err != nil {
+			return resource.NonRetryableError(err)
+		}
+
+		return nil
+	})
+
+	if tfresource.TimedOut(err) {
+		output, err = FindQueueAttributesByURL(conn, d.Id())
+	}
```
Comment (ewbankkit) [inline internal/service/sqs/queue.go:306]: Alternatively:

```go
	outputRaw, err := tfresource.RetryWhenNotFound(queueReadTimeout, func() (interface{}, error) {
		return FindQueueAttributesByURL(conn, d.Id())
	})

	// Error checking...

	output := outputRaw.(map[string]string)
```

File: internal/service/sqs/queue_policy.go [hunk 3]
```diff
@@ -83,7 +96,30 @@ func resourceQueuePolicyUpsert(d *schema.ResourceData, meta interface{}) error {
...
+		return nil
+	})
+
+	if tfresource.TimedOut(err) {
+		policy, err = FindQueuePolicyByURL(conn, d.Id())
+	}
+
+	if err != nil {
+		return fmt.Errorf("error reading SQS Queue Policy (%s): %w", d.Id(), err)
+	}
```
Comment (ewbankkit) [inline internal/service/sqs/queue_policy.go:122]: As for `aws_sqs_queue`, remove (there is an identical code block below).

File: internal/service/sqs/queue_policy.go [hunk 4]
```diff
@@ -83,7 +96,30 @@ func resourceQueuePolicyUpsert(d *schema.ResourceData, meta interface{}) error {
...
+		if err != nil {
+			return resource.NonRetryableError(err)
+		}
+
+		return nil
+	})
+
+	if tfresource.TimedOut(err) {
+		policy, err = FindQueuePolicyByURL(conn, d.Id())
+	}
```
Comment (ewbankkit) [inline internal/service/sqs/queue_policy.go:118]: Alternatively:

```go
	outputRaw, err := tfresource.RetryWhenNotFound(queueReadTimeout, func() (interface{}, error) {
		return FindQueuePolicyByURL(conn, d.Id())
	})

	// Error checking...

	policy := outputRaw.(string)
```

**Comment by joe-a-t** [comment]
This still seems to be an issue with https://github.com/hashicorp/terraform-provider-aws/issues/22197#issuecomment-1009162524 even after upgrading to 3.71.0 so we have had to pin to 3.68.0

**Comment by zacblazic** [comment]
I'm using v3.71.0 and seeing whitespace diffs for the `policy` argument when setting it via `jsonencode()`.

**Pull request merged**